File: 0448-find-all-numbers-disappeared-in-an-array/0448-find-all-numbers-disappeared-in-an-array.py
class Solution:
    def findDisappearedNumbers(self, nums: List[int]) -> List[int]:
        
        res = []
        for num in nums:
            index = abs(num) - 1
            nums[index] = -abs(nums[index])
            
        for i,n in enumerate(nums):
            if n > 0:
                res.append(i+1)
        return res

File: 0448-find-all-numbers-disappeared-in-an-array/0448-find-all-numbers-disappeared-in-an-array.java
class Solution {
    public List<Integer> findDisappearedNumbers(int[] nums) {
        ArrayList<Integer> res = new ArrayList<>();
        int n = nums.length;

        for(int i=0; i<n; i++){
            int index = Math.abs(nums[i]) - 1;
            nums[index] = -Math.abs(nums[index]);
        }

        for(int i=0; i<n; i++){
            if (nums[i] > 0){
                res.add(i+1);
            }

        } 

        return res;
        
    }
}